### DISCUSSION TITLE
graphscope源码阅读和编译

### DISCUSSION THREAD

**Discussion by Maxismcaptain** [alibaba/GraphScope#3742] [Q&A]
请问graphscope源码阅读有推荐的顺序吗，各个模块间的主要源码是哪些文件夹呢，另外就是如果要对源码做一些修改，比如简单的加一句print（"hello world"） 之后如何编译呢，又如何调用刚才编译的代码来看到输出的hello world

**Comment by yecol** [comment]
Hi @Maxismcaptain 

Thanks for your interests in GraphScope！
A good start is from https://graphscope.io/docs/overview/intro

The main components include
- python client: `/python` 
- `interactive_engine`
- `analytical_engine`
- ...

An easy coding practise can begin with the python client. :)
